vertical-pod-autoscaler/pkg/updater/eviction/pods_eviction_restriction.go

```diff
@@ -156,6 +159,61 @@ func (e *podsEvictionRestrictionImpl) Evict(podToEvict *apiv1.Pod, eventRecorder
 	return nil
 }
 
+// canDelete checks if the container is in a CrashLoopBackOff state and if it was OOMKilled.
+func (e *podsEvictionRestrictionImpl) canDelete(containerStatus apiv1.ContainerStatus) bool {
+	return containerStatus.LastTerminationState.Terminated != nil &&
+		containerStatus.LastTerminationState.Terminated.Reason == "OOMKilled" &&
+		containerStatus.State.Waiting != nil &&
+		containerStatus.State.Waiting.Reason == "CrashLoopBackOff"
+}
+
+// EvictViaDelete sends deletion instruction to api client. Returns error if pod cannot be deleted or if client returned error
+// Does not check if pod was actually deleted after termination grace period.
+func (e *podsEvictionRestrictionImpl) EvictViaDelete(podToEvict *apiv1.Pod, eventRecorder record.EventRecorder) error {
+	cr, present := e.podToReplicaCreatorMap[getPodID(podToEvict)]
```

Comment:
At first I was confused why we set `cr` here but use it much later. Please add a comment like

// Make sure that we can map pod to replica so any problems prevent deleting the pod.